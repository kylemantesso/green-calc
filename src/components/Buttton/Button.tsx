import * as React from "react";

import styled from "styled-components/native";
import {Colours} from "../../style/Colours";

export const DIGITS = ["0","1","2","3","4","5"];
export const OPERATORS = ["=","+","-","/","X","C"];

export type ButtonLabelType = typeof DIGITS[number] | typeof OPERATORS[number] ;

const ButtonElement = styled.TouchableOpacity`
  align-items: center;
  padding: 10px;
  flex: 1;
  border: 5px ${Colours.DARK_GREEN} solid;
  border-radius: 10px;
  margin: 4px;
  background: ${({selected}: {selected?: boolean}) => selected ? Colours.GREEN_HIGHLIGHT : Colours.WHITE };
`;

const ButtonLabel = styled.Text`
  font-size: 38px;
  color: ${Colours.GREY};
`;

interface Props {
  label: ButtonLabelType;
  handler: (buttonLabel: ButtonLabelType) => void;
  selected?: boolean;
}

export const Button = ({ handler, label, selected }: Props) => (
  <ButtonElement selected={selected} onPress={() => handler(label)}>
    <ButtonLabel>{label}</ButtonLabel>
  </ButtonElement>
);

