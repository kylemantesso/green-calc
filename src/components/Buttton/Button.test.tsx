import React from "react";

import renderer from "react-test-renderer";
import {Button} from "./Button";

describe('Button', () => {

});

test("renders correctly", () => {
  const tree = renderer.create(<Button label={"1"} selected  handler={() => undefined}/>).toJSON();
  expect(tree).toMatchSnapshot();
});
