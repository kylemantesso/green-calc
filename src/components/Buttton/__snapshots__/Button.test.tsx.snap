// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#f4ff61",
      "borderColor": "#27aa80",
      "borderRadius": 10,
      "borderStyle": "solid",
      "borderWidth": 5,
      "flexBasis": 0,
      "flexGrow": 1,
      "flexShrink": 1,
      "marginBottom": 4,
      "marginLeft": 4,
      "marginRight": 4,
      "marginTop": 4,
      "opacity": 1,
      "paddingBottom": 10,
      "paddingLeft": 10,
      "paddingRight": 10,
      "paddingTop": 10,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#333",
          "fontSize": 38,
        },
      ]
    }
  >
    1
  </Text>
</View>
`;
